Controller/Adminhtml/CompanyCodes/Get.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @category    ClassyLlama
+ * @copyright   Copyright (c) 2018 Classy Llama Studios, LLC
+ * @author      sean.templeton
+ */
+
+namespace ClassyLlama\AvaTax\Controller\Adminhtml\CompanyCodes;
+
+use ClassyLlama\AvaTax\Exception\AvataxConnectionException;
+use Magento\Framework\DataObject;
+
+class Get extends \Magento\Backend\App\Action
+{
+    protected $resultPageFactory;
+
+    /**
+     * @var \ClassyLlama\AvaTax\Framework\Interaction\Rest\Company
+     */
+    protected $company;
+
+    /**
+     * @var \ClassyLlama\AvaTax\Helper\Config
+     */
+    private $config;
+
+    public function __construct(
+        \Magento\Backend\App\Action\Context $context,
+        \Magento\Framework\Controller\Result\JsonFactory $resultPageFactory,
+        \ClassyLlama\AvaTax\Framework\Interaction\Rest\Company $company,
+        \ClassyLlama\AvaTax\Helper\Config $config
+    )
+    {
+        parent::__construct( $context );
+        $this->resultPageFactory = $resultPageFactory;
+        $this->company = $company;
+        $this->config = $config;
+    }
+
+    public function execute()
+    {
+        $companies = [];
+        $postValue = $this->getRequest()->getPostValue();
+        $isProduction = (bool)$this->getRequest()->getParam( 'mode' );
+        $resultJson = $this->resultPageFactory->create();
+        $scope = isset( $postValue['scope'] ) ? $postValue['scope'] : null;
+        $scopeType = $postValue['scope_type'] === 'global' ? \Magento\Store\Model\ScopeInterface::SCOPE_STORE : $postValue['scope_type'];
+        $currentCompanyId = $this->config->getCompanyId( $scope, $scopeType, $isProduction );
+
+        try
+        {
+            if (!isset( $postValue['license_key'] ))
+            {
+                $postValue['license_key'] = $this->config->getLicenseKey( $scope, $scopeType, $isProduction );
+            }
+
+            $companies = $this->company->getCompaniesWithSecurity(
+                $postValue['account_number'],
+                $postValue['license_key'],
+                null,
+                $isProduction
+            );
+        }
+        catch (AvataxConnectionException $e)
```

Comment:
@LordZardeck - As a note, any time we have an empty catch it should have an explicit comment explaining why it is empty.  In this case, though, I don't really think it's sufficient to have the empty catch.  If there's an AvaTax connection error, instead of just resulting in an empty drop-down of companies, we really do need to throw an error and have the JS be capable of catching and displaying it to the user.

Reply:
@LordZardeck - This comment was not addressed